Lib/unittest/mock.py

```diff
@@ -2203,9 +2496,26 @@ def create_autospec(spec, spec_set=False, instance=False, _parent=None,
                  name=_name, **_kwargs)
 
     if isinstance(spec, FunctionTypes):
+        wrapped_mock = mock
         # should only happen at the top level because we don't
         # recurse for functions
         mock = _set_signature(mock, spec)
+        if is_async_func:
+            mock._is_coroutine = _is_coroutine
```

Comment:
I think `_is_coroutine` should be `asyncio.coroutines._is_coroutine`. Personally I think the attribute setup logic could be moved out of `create_autospec` as a utility function and would also help in backport where the setup function could just be commented.

```python
def test_create_autospec(self):
    spec = create_autospec(async_func)
```

```
======================================================================
ERROR: test_create_autospec (Lib.unittest.test.testmock.testasync.AsyncAutospecTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/karthikeyansingaravelan/stuff/python/cpython/Lib/unittest/test/testmock/testasync.py", line 156, in test_create_autospec
    spec = create_autospec(async_func)
  File "/Users/karthikeyansingaravelan/stuff/python/cpython/Lib/unittest/mock.py", line 2504, in create_autospec
    mock._is_coroutine = _is_coroutine
NameError: name '_is_coroutine' is not defined
```

----------------------------------------------------------------------

Reply:
I think you are right, this looks like a mistake, thanks for catching! I agree the attribute setup can be moved out into a utility function, let's do it in another PR.